### ISSUE TITLE
Support variabilisation of docker base image

### ISSUE THREAD

**Issue by LvffY** [hadolint/hadolint#858]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

## Context

Because we may use multistage docker image and just want to use a variable to store our base image name, we would want hadolint to analyse correctly our base image and the possible variables used in the process. 

### Expected behavior

It would be great 

My report should be clear.

### Actual behavior

My report contains the error `DL3006 : warning: Always tag the version of an image explicitly`.

### Steps to reproduce the behavior

Run with the latest docker image : 

```bash
docker run --rm -i hadolint/hadolint < Dockerfile
```

**Dockerfile**

```Dockerfile
# syntax=docker/dockerfile:1
ARG BASE_IMAGE=alpine:3

FROM ${BASE_IMAGE} as main
```

As a workaround to avoid the warning, you can have the following docker file :

```Dockerfile
# syntax=docker/dockerfile:1
ARG BASE_IMAGE=alpine

FROM ${BASE_IMAGE}:3 as main
```

**Additional environment details (OS, stack version, etc.)**
